File: 0228-summary-ranges/0228-summary-ranges.py
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        if not nums: return list()

        result = list()
        start = nums[0]

        for i in range(1, len(nums)):
            if nums[i] == nums[i - 1] + 1: continue
            else:
                if start == nums[i - 1]: result.append(str(start))
                else: result.append(f'{start}->{nums[i - 1]}')
                start = nums[i]

        if start == nums[-1]: result.append(str(start))
        else: result.append(f'{start}->{nums[-1]}')

        return result

File: 0228-summary-ranges/0228-summary-ranges.java
class Solution {
    public List<String> summaryRanges(int[] nums) {
        if (nums == null || nums.length == 0) {
            return new ArrayList<>();
        }

        List<String> result = new ArrayList<>();
        int start = nums[0];

        for (int i = 1; i < nums.length; i++) {
            if (nums[i] == nums[i - 1] + 1) {
                continue;
            }
            if (start == nums[i - 1]) {
                result.add(Integer.toString(start));
            } else {
                StringBuilder sb = new StringBuilder();
                sb.append(Integer.toString(start));
                sb.append("->");
                sb.append(Integer.toString(nums[i - 1]));
                result.add(sb.toString());
            }
            start = nums[i];
        }

        if (start == nums[nums.length - 1]) {
            result.add(Integer.toString(start));
        } else {
            StringBuilder sb = new StringBuilder();
            sb.append(Integer.toString(start));
            sb.append("->");
            sb.append(Integer.toString(nums[nums.length - 1]));
            result.add(sb.toString());
        }

        return result;
    }
}